03-task-manager/starter/controllers/ejemploController.js

```diff
@@ -1,43 +1,30 @@
 const Example = require('../models/ejemplo');
+const asyncWrapper = require('../middleware/async');
+const { createCustomError } = require('../errors/custom-error');
 
 // Controller to create a new example
-exports.createExample = async (req, res) => {
-  try {
+exports.createExample = asyncWrapper (async (req, res) => {
     const example = new Example(req.body);
     const result = await example.save();
     res.json(result);
-  } catch (error) {
-    console.log(error);
-    res.status(500).send(error);
-  }
-};
+});
 
 // Controller to get the last names of all people
-exports.getLastNames = async (req, res) => {
-  try {
+exports.getLastNames = asyncWrapper (async (req, res) => {
     const examples = await Example.find({}, 'nombre');
     const lastNames = examples.map((e) => e.nombre.split(' ').pop());
     res.json(lastNames);
-  } catch (error) {
-    console.log(error);
-    res.status(500).send(error);
-  }
-};
+});
 
 // Controller to convert to "Title Case"
-exports.convertToTitleCase = async (req, res) => {
-  try {
+exports.convertToTitleCase = asyncWrapper (async (req, res) => {
     const examples = await Example.find({}, 'nombre');
     const titleCaseNames = examples.map((e) => toTitleCase(e.nombre));
     res.json(titleCaseNames);
-  } catch (error) {
-    res.status(500).send(error);
-  }
-};
-
-exports.updateFullName = async (req, res) => {
-  try {
-    const { _id } = req.params;
+});
+
+exports.updateFullName = asyncWrapper (async (req, res) => {
```

Comment:
This controller doesn't work for me.  Try referencing the code in /03-task-manager/final/controller/tasks.js to see how the instructor creates an update function.